spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/StreamListenerTestMessageArgument.java

```diff
@@ -0,0 +1,114 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.config;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Output;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.handler.annotation.SendTo;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(Parameterized.class)
+public class StreamListenerTestMessageArgument {
+
+	private Class<?> configClass;
+
+	public StreamListenerTestMessageArgument(Class<?> configClass) {
+		this.configClass = configClass;
+	}
+
+	@Parameterized.Parameters
+	public static Collection InputConfigs() {
+		return Arrays.asList(new Class[] {TestPojoWithMessageArgument1.class, TestPojoWithMessageArgument2.class});
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testMessageArgument() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication
+				.run(this.configClass, "--server.port=0");
+		MessageCollector collector = context.getBean(MessageCollector.class);
+		Processor processor = context.getBean(Processor.class);
+		String id = UUID.randomUUID().toString();
+		processor.input().send(MessageBuilder.withPayload("barbar" + id)
+				.setHeader("contentType", "text/plain").build());
+		TestPojoWithMessageArgument testPojoWithMessageArgument = context
+				.getBean(TestPojoWithMessageArgument.class);
+		assertThat(testPojoWithMessageArgument.receivedMessages).hasSize(1);
+		assertThat(testPojoWithMessageArgument.receivedMessages.get(0).getPayload()).isEqualTo("barbar" + id);
+		Message<StreamListenerTestInterfaces.BarPojo> message = (Message<StreamListenerTestInterfaces.BarPojo>) collector
+				.forChannel(processor.output()).poll(1, TimeUnit.SECONDS);
+		assertThat(message).isNotNull();
+		assertThat(message.getPayload().getBar()).isEqualTo("barbar" + id);
+		context.close();
+	}
+
+	@EnableBinding(Processor.class)
+	@EnableAutoConfiguration
+	public static class TestPojoWithMessageArgument1 extends TestPojoWithMessageArgument {
```

Comment:
again TestPojoWithMessageArgument1 and TestPojoWithMessageArgument2 appear to be identical. What's the point?